### PR TITLE
Fix/refactor client

### PR DESCRIPTION
**Pull request by mbeilin**
In this PR:

-  added `json.dupms()` wraper to request body, like:

```
body = json.dumps({
                'operation': 'upload',
                'transfers': ['basic'],
                'ref': { 'name': 'refs/heads/contrib' },
                'objects': [
                    {
                    'oid': file_hash,
                    'size': file_size,
                    },
                ],
            })
``` 

- removed unnecessary comma after `headers` that caused a run time error:
```
...
headers = {
            'Accept': 'application/vnd.git-lfs+json',
            'Content-Type': 'application/vnd.git-lfs+json',
            'Authorization': 'Bearer {}'.format(jwt_auth_token),
        } ,
```

### CODE REVIEW

**Comment by rufuspollock** [comment]
@cuducos please can you review this asap

**Review by cuducos — APPROVED** [review]
The usage of  `json.dumps` is not needed. What we need is to pass `body` to the `json` keyword-argument instead of `data`, then `requests` handles the serialization — but this is a possible enhancement for a minor commit, definitively not a blocker to the PR.

Nice catch regarding the `,` accidentally transforming `headers` in a tuple!

**Comment by cuducos** [comment]
By the way, as #18 was merged, these bugs were probably fixed there, so I am not sure whether we need to merge this PR. I don't think so.

**Comment by mbeilin** [comment]
> By the way, as #18 was merged, these bugs were probably fixed there, so I am not sure whether we need to merge this PR. I don't think so.

@cuducos I will test it with #18  merged and will update you. Probably we really don't need this #19 to be merged anymore.
Anyway, that was a massive change here to all the code stack in #18, so I think #19 is not relevant anymore.

**Comment by cuducos** [comment]
Yes, the previous code had no abstractions by responsibility, and I suggested that — moving different parts of the CKAN client to purpose-specific clients (GitLab, Git, calling command-line apps, etc.). It is hard to follow the diff, I know and I am sorry about that. But I do hope it enhances maintainability in the medium term.   

On top of that, I must say that I couldn't test `push_blob` properly (there were a few lines on the PR regarding this issue), so feel free to reach me if you find this error, solutions, or anything else  : ) Many thanks

**Comment by mbeilin** [comment]
> Yes, the previous code had no abstractions by responsibility, and I suggested that — moving different parts of the CKAN client to purpose-specific clients (GitLab, Git, calling command-line apps, etc.). It is hard to follow the diff, I know and I am sorry about that. But I do hope it enhances maintainability in the medium term.
> 
> On top of that, I must say that I couldn't test `push_blob` properly (there were a few lines on the PR regarding this issue), so feel free to reach me if you find this error, solutions, or anything else : ) Many thanks

@cuducos so started form the `push_blob`, 
```
from frictionless import describe
resource = describe('ckan-client-py/tests/sample_file/dailyprices.csv')
res = ckan_uploader.push_blob(resource)
```
getting the following error:
```
Traceback (most recent call last):
  File "/snap/pycharm-community/211/plugins/python-ce/helpers/pydev/pydevd.py", line 1448, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/snap/pycharm-community/211/plugins/python-ce/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/home/michael/PycharmProjects/nhs_upload_test/ckan-client-py/main.py", line 33, in <module>
    upload_resource()
  File "/home/michael/PycharmProjects/nhs_upload_test/ckan-client-py/main.py", line 29, in upload_resource
    res = ckan_uploader.push_blob(resource)
  File "/home/michael/PycharmProjects/nhs_upload_test/ckan-client-py/ckanclient/__init__.py", line 165, in push_blob
    lfs = self.auth.request_file_upload_actions(resource)
  File "/home/michael/PycharmProjects/nhs_upload_test/ckan-client-py/ckanclient/auth.py", line 55, in request_file_upload_actions
    'oid': resource['descriptor']['hash'],
KeyError: 'descriptor'
```
BTW, can you provide me with some instructions, like it was in previous README version? - https://github.com/datopian/ckan-client-py/tree/fix/refactor_client#developers
For example, I need to create a new Resource in the existing Dataset, what are my steps?
First, `push_blob` for upload file for this resource and then?
Thanks!

**Comment by cuducos** [comment]
> getting the following error:

Good news: I am working on that ATM!
Bad news: I am still working on that…

> BTW, can you provide me with some instructions, like it was in previous README version? 

Can you check if what was added to CKAN Client Guide is a good reference? (I mean, assuming I fix `push_blob`) https://github.com/datopian/tech.datopian.com/pull/24
